include/pybind11/iostream.h

```diff
@@ -67,8 +72,8 @@ class pythonbuf : public std::streambuf {
     pythonbuf(pythonbuf&&) = default;
 
     /// Sync before destroy
-    ~pythonbuf() {
-        sync();
+    ~pythonbuf() override {
+        _sync();
```

Comment:
No, you shouldn't call virtual functions anymore in a destructor, indeed. But you should be able to just call `pythonbuf::sync` instead, no? Isn't that making things simpler ánd more clear than this workaround?